### PR TITLE
feat: campaign groups in popup action card

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds UI to the Campaigns Wizard to control the Campaign Groups assigned to a Campaign. 

Dependent PRs:
~https://github.com/Automattic/newspack-plugin/pull/764~

This branch must be tested along with https://github.com/Automattic/newspack-popups/pull/363

Screen Shot 2020-12-30 at 6 32 09 PM

### How to test the changes in this Pull Request:

Switch `newspack-popups` to `feat/campaign-groups-in-popup-action-card`.

1. Make sure there are some Campaign Groups defined. This can be done by editing any campaign and adding campaign groups in the pane near the bottom of the sidebar. Note: at a later point we'll add UI to create Campaign Groups directly in the Campaigns wizard but this would be premature.
2. In the Campaigns wizard, click the hamburger icon on any Campaign. Observe a `Campaign Groups` autocomplete field.
3. Begin typing one of the campaign group names, observe autocomplete finds it. Click the name and observe it is added.
4. Refresh the page and observe the assigned campaign groups remain. 
5. Try removing a campaign group and observe this works as expected.
6. For Sitewide default Overlay Campaigns, observe that the hamburger tooltip reads `Campaign groups` instead of `Category filtering and campaign groups`. Click the hamburger and observe that only the Campaign Groups UI is visible.
7. Verify that the `Category filtering` fields work as they always have.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/popups/components/popup-action-card/index.js [hunk 1]
```diff
@@ -74,12 +85,22 @@ const PopupActionCard = ( {
...
-				<CategoryAutocomplete
-					value={ categories || [] }
-					onChange={ tokens => setCategoriesForPopup( id, tokens ) }
-					label={ __( 'Category filtering', 'newspack ' ) }
-					disabled={ sitewideDefault }
-				/>
+				<Fragment>
+					<CategoryAutocomplete
+						value={ campaignGroups || [] }
+						onChange={ tokens => setTermsForPopup( id, tokens, 'newspack_popups_taxonomy' ) }
```
Comment (dkoo) [inline assets/wizards/popups/components/popup-action-card/index.js:91]: Nitpick, but this is set as a constant `Newspack_Popups::NEWSPACK_POPUPS_TAXONOMY` in that repo. Could we localize the taxonomy name using the constant so it's always in sync here?

**Review by dkoo — CHANGES_REQUESTED** [review]
Works well as described, just have one nit that might help with code maintenance across repos.

File: assets/wizards/popups/components/popup-action-card/index.js (hunk 1 above)
Comment (jeffersonrabb) [reply assets/wizards/popups/components/popup-action-card/index.js]: Addressed in https://github.com/Automattic/newspack-plugin/pull/767/commits/fb4cd9a522a99a9f45dbf67db43064bfe3e7a0ac and https://github.com/Automattic/newspack-popups/pull/363/commits/ac4441efb365f4a5684ba597319f6e4a8d9e3ff7

**Review by dkoo — APPROVED** [review]
Latest change in fb4cd9a is exactly what I had in mind. Approved as soon as we fix the linting error:

```
/home/circleci/project/assets/wizards/popups/components/popup-action-card/index.js
  93:27  error  Replace `·setTermsForPopup(·id,·tokens,·window.newspack_popups_wizard_data.taxonomy·)·` with `⏎↹↹↹↹↹↹↹setTermsForPopup(·id,·tokens,·window.newspack_popups_wizard_data.taxonomy·)⏎↹↹↹↹↹↹`  prettier/prettier
```

**Comment by jeffersonrabb** [comment]
> Latest change in [fb4cd9a](https://github.com/Automattic/newspack-plugin/commit/fb4cd9a522a99a9f45dbf67db43064bfe3e7a0ac) is exactly what I had in mind. Approved as soon as we fix the linting error:
> 
> ```
> /home/circleci/project/assets/wizards/popups/components/popup-action-card/index.js
>   93:27  error  Replace `·setTermsForPopup(·id,·tokens,·window.newspack_popups_wizard_data.taxonomy·)·` with `⏎↹↹↹↹↹↹↹setTermsForPopup(·id,·tokens,·window.newspack_popups_wizard_data.taxonomy·)⏎↹↹↹↹↹↹`  prettier/prettier
> ```

Fixed in https://github.com/Automattic/newspack-plugin/pull/767/commits/6187cae1b665e7443ddded5f83e4c338c1fa9608

**Pull request merged**